Remotion/Web/Development.WebTesting/RequestErrorDetectionStrategies/AspNetRequestErrorDetectionParser.cs

```diff
@@ -71,33 +86,54 @@ public string Stacktrace
       }
     }
 
-    public Result Parse ([NotNull] ElementScope scope)
-    {
-      ArgumentUtility.CheckNotNull("scope", scope);
+    private const string c_errorPageDetectionJs = @"
+function fixNewlines(text) {
+  // Ensure strings have a CRLF endings
+  return text.replaceAll(/(?<!\r)\n/g, '\r\n');
+}
 
-      var pageRequestManagerException = scope.FindCss(".SmartPageErrorBody:first-child", Options.NoWait);
+let result = null;
 
-      var startSelector = GetRootSelectorForErrorInformation(pageRequestManagerException);
+const smartPageErrorBody = document.querySelector('.SmartPageErrorBody:first-child');
```

Comment:
considering what we discovered about performance, we can improve this initial detection:
`body > div#SmartPageServerErrorMessage` first and then drill into `:scope .SmartPageErrorBody:first-child > div` to get the `errorPageTarget` for async requests.

Please verify to make sure I'm right about that :)

Reply:
Right now the detection takes 2ms, which includes the JS execution + JSON deserialization. I don't think it is worth trying to optimize anymore as it might change the existing behavior.

Reply:
Did you measure the 2ms in async, too? Also with a big (deep) page or just demo pages. Thing is, when there's a ton of stuff on the page, things might look different. Anyway, the updated version would be much more precise/safer actually since we detect & capture based on the root element and only then select what we need.